numpy/core/shape_base.py

```diff
@@ -187,22 +179,105 @@ def atleast_3d(*arys):
     [[[1 2]]] (1, 1, 2)
 
     """
-    res = []
-    for ary in arys:
-        ary = asanyarray(ary)
-        if ary.ndim == 0:
-            result = ary.reshape(1, 1, 1)
-        elif ary.ndim == 1:
-            result = ary[_nx.newaxis, :, _nx.newaxis]
-        elif ary.ndim == 2:
-            result = ary[:, :, _nx.newaxis]
-        else:
-            result = ary
-        res.append(result)
-    if len(res) == 1:
-        return res[0]
-    else:
-        return res
+    return _atleast_nd_result(_atleast_nd_result(arys, 2, 0, False), 3, -1)
```

Comment:
This would become
```suggestion
    return _unpack(atleast_nd(atleast_nd(a, 2, 0), 3, -1)) for a in args)
```
with my suggestion above

Reply:
Each call to `atleast_nd` requires a separate comprehension. It only accepts one argument. This was determined to be the best way in the previous PR. I happen to agree.

Reply:
I don't get what you mean - within the comprehension, `a` is just an array

Reply:
I've been staring at this too long. You're absolutely right.